lib/runner/replay-controller.js

```diff
@@ -55,45 +54,33 @@ _.assign(ReplayController.prototype, /** @lends ReplayController.prototype */{
             // abortOnError makes sure request command bubbles errors
             // so we can pass it on to the callback
             abortOnError: true,
-            // @todo: need a new identifier for replay vs intermediate request?
-            auth: waitForResponse ? null : context.auth,
             replayState: context.replayState
         };
 
-        // if waitForResponse is true, current execution waits for the response
-        // callback is invoked only after the response is received
-        if (waitForResponse) {
-            this.run.immediate('request', payload)
-                .done(function (response) {
-                    success(null, response);
-                })
-                .catch(function (err) {
-                    success(err);
-                });
-        }
-
-        // or schedule a request and continue with current queue
-        else {
-            this.run.interrupt('request', payload);
-            success();
-        }
+        this.run.immediate('httpRequest', payload)
+            .done(function (response) {
+                success(null, response);
+            })
+            .catch(function (err) {
```

Comment:
Shouldn't this be a direct reference to `failure`?

Reply:
If you are talking about `success` vs `failure`, here `success` is when `ReplayController` decides to send a request. `failure` is when it decided not to replay the request due to max retries exceeding and not the failure in the actual request.

Reply:
Either way, `success(err);` looks weird.